### PR TITLE
Added .github/workflows/lint.yml

### LINKED ISSUE #3519: Create a workflow detect EOF and fix it for PR automatically
### Summary

Seems linelint[1] can help us to do this.

### Appendix

1. https://github.com/marketplace/actions/linelint

### PR DESCRIPTION
**Pull request by arujjval**
closes #3519

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# .github/workflows/main.yml
```
Comment (0yi0) [inline .github/workflows/lint.yml:1]: Why this?

**Review by 0yi0 — CHANGES_REQUESTED** [review]
```
[EOF Rule] File ".github/workflows/lint.yml" has lint errors
```
Fix the comment and lint error for the file itself and I think it's ready to merge.

**Comment by 0yi0** [comment]
Oh I see the original issue is asking for [autofix](https://github.com/marketplace/actions/linelint#autofix), so please also set autofix to true. No need to fix the lint error manually then.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (arujjval) [reply .github/workflows/lint.yml:1]: It was there with default snippet. Should I remove it?

Comment (0yi0) [reply .github/workflows/lint.yml:1]: Or change it to something actually describing this file, comment should be meaningful.

**Comment by hydai** [comment]
The name `lint` is too general. Use `linelint` instead.

File: .github/workflows/linelint.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# .github/workflows/main.yml
```
Comment (hydai) [inline .github/workflows/linelint.yml:1]: This line is redundant. Since `linelint.yml` is added, `lint.yml` should be removed.

**Review by hydai — CHANGES_REQUESTED** [review]

File: .linelint.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+ignore:
+  - .git/
+
+rules:
+  end-of-file:
+    enable: true
+    disable-autofix: false
+    ignore:
+      - README.md
```
Comment (hydai) [inline .linelint.yml:11]: What's the reason to ignore all README files?

**Review by hydai — CHANGES_REQUESTED** [review]

File: .linelint.yml (hunk 3 above)
Comment (arujjval) [reply .linelint.yml:11]: No apparent reason.

**Comment by arujjval** [comment]
@hydai for some reason the auto fixing is not working. Is there any way I can test Github actions locally?

File: .linelint.yml (hunk 3 above)
Comment (arujjval) [reply .linelint.yml:11]: I will remove it then.

**Comment by hydai** [comment]
> @hydai for some reason the auto fixing is not working. Is there any way I can test Github actions locally?

You can run the actions inside your cloned repo.

**Comment by arujjval** [comment]
@hydai  I tested it in my forked repo. It seems it's autofixing buggy. I have created an appropriate configuration file to achieve that but it's not working. Can I look into other linting options to achieve this task?

**Comment by dannypsnl** [comment]
Sure

**Comment by arujjval** [comment]
I am making a new PR.

**Pull request closed without merging**